lib/mainloop-control.c

```diff
@@ -34,37 +34,41 @@
 
 #include <string.h>
 
+static gboolean
+_control_process_log_level(const gchar *level, GString *result)
+{
+  if (!level)
+    return FALSE;
+  gint ll = msg_map_string_to_log_level(level);
+  if (ll < 0)
+    return FALSE;
+
+  msg_set_log_level(ll);
+  return TRUE;
+}
 static void
 control_connection_message_log(ControlConnection *cc, GString *command, gpointer user_data, gboolean *cancelled)
 {
   gchar **cmds = g_strsplit(command->str, " ", 3);
-  gboolean on;
   GString *result = g_string_sized_new(128);
+  gboolean success;
 
   if (!cmds[1])
     {
       g_string_assign(result, "FAIL Invalid arguments received");
       goto exit;
     }
 
-  gint ll = msg_map_string_to_log_level(cmds[1]);
-  if (ll >= 0)
-    {
-      if (cmds[2])
-        {
-          on = g_str_equal(cmds[2], "ON");
-          if (!on)
-            ll = ll - 1;
+  gint orig_log_level = msg_get_log_level();
 
-          gint orig_log_level = msg_get_log_level();
-          msg_set_log_level(ll);
+    success = _control_process_compat_log_command(cmds[1], cmds[2], result);
```

Comment:
This patch can not be compiled, I think the introduction of `_control_process_compat_log_command()` and `_control_process_log_level` is mixed up here and the next patch (`mainloop-control: extract LOG <level> <onoff> processing into a separ… `).

Reply:
the rebase of the branch should help here. I indeed split a patch incorrectly.